keps/sig-scheduling/5075-dra-consumable-capacity/README.md

```diff
@@ -0,0 +1,1533 @@
+}
+
+// ExactDeviceRequest is a request for one or more identical devices.
+type ExactDeviceRequest struct {
+...
+   // CapacityRequests define resource requirements against each capacity.
+   //
+   // +optional
+   // +featureGate=DRAConsumableCapacity
+   CapacityRequests *CapacityRequirements
+}
+
+
+// CapacityRequirements defines the capacity requirements for a specific device request.
+type CapacityRequirements struct {
+    // Minimum defines the minimum amount of each device capacity required for the request.
+    // Each minimum amount must be a standard Quantity.
+    //
+    // If the capacity has a sharing policy, this value is rounded up to the nearest valid amount
+    // according to that policy. The rounded value is used during scheduling to determine how much capacity to consume.
+    //
+    // If the quantity does not have a sharing policy, this value is used as an additional filtering
+    // condition against the available capacity on the device.
+    // This is semantically equivalent to a CEL selector with
+    // `device.capacity[<domain>].<name>.compareTo(quantity(<minimum quantity>)) >= 0`
+    // For example, device.capacity['test-driver.cdi.k8s.io'].counters.compareTo(quantity('2')) >= 0
+    //
+    // +optional
+   Minimum map[QualifiedName]resource.Quantity
+}
+
+// DeviceConstraint must have exactly one field set besides Requests.
+type DeviceConstraint struct {
+	...
+   // DistinctAttribute requires that all devices in question have this
+   // attribute and that its type and value are unique across those devices.
+   //
+   // This acts as the inverse of MatchAttribute.
+   //
+   // This constraint is used to avoid allocating multiple requests to the same device
+   // by ensuring attribute-level differentiation.
+   //
+   // This is useful for scenarios where resource requests must be fulfilled by separate physical devices.
+   // For example, a container requests two network interfaces that must be allocated from two different physical NICs.
+   //
+   // +optional
+   // +oneOf=ConstraintType
+   DistinctAttribute *FullyQualifiedName
+}
+
+```
+
+#### ResourceClaimStatus's DeviceRequestAllocationResult
+
+```go
+type DeviceRequestAllocationResult struct {
+  ...
+
+	// ShareID uniquely identifies an individual allocation share of a device,
+	// used when the device supports multiple simultaneous allocations.
+	// It serves as an additional map key to differentiate concurrent shares
+	// of the same device.
+	//
+	// The ID is randomly generated as a hexadecimal (hex) string for each allocation share.
+	// It must be unique among all currently allocated shares for the same device
+	// (i.e., uniqueness is required only at the device level, not globally).
+	//
+	// Hex is chosen for its compact representation, ease of generation from binary,
+	// and suitability for identifiers in logs, APIs, and storage.
+	//
+	// +optional
+	// +featureGate=DRAConsumableCapacity
+	ShareID *string
```

Comment:
Nit: indent with the same amount of spaces as elsewhere.

I'm undecided whether we want to enforce this as a hex value. This seems a bit unusual from an API perspective and, as presented here, doesn't specify a maximum length of the string.

Perhaps just go with the usual "must be a DNS label"? Doing it as a short hex string then becomes an implementation detail, without being part of the API.

Let's first clarify that before re-indenting, otherwise this comment thread is immediately "obsolete"...